File: Algorithms/0658_Find_K_Closest_Elements/Python/Find_K_Closest_Elements_Solution_2.py

# Space: O(n)
# Time: O(n)

class Solution:
    def findClosestElements(self, arr, k, x):
        length = len(arr)
        if k >= length: return arr
        sorted_arr = sorted(arr, key=lambda temp: abs(temp - x))
        return sorted(sorted_arr[:k])

File: Algorithms/0658_Find_K_Closest_Elements/Java/FindKClosestElementsSolution2.java

// Space: O(n)
// Time: O(n)

class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        List<Integer> res = new ArrayList<>();
        for (int i : arr) {
            res.add(i);
        }
        res.sort((a, b) -> Math.abs(a - x) - Math.abs(b - x));
        //res.sort(Comparator.comparingInt(a -> Math.abs(a - x)));
        res = res.subList(0, k);
        Collections.sort(res);
        return res;
    }
}




